sidebar.c

```diff
@@ -294,10 +294,13 @@ static const char *sidebar_format_str(char *buf, size_t buflen, size_t col, int
 static void make_sidebar_entry(char *buf, size_t buflen, int width,
                                const char *box, struct SbEntry *sbe)
 {
-  if (!buf || !box || !sbe)
+  if (!buf)
     return;
 
-  mutt_str_strfcpy(sbe->box, box, sizeof(sbe->box));
+  if (box && sbe)
+    mutt_str_strfcpy(sbe->box, box, sizeof(sbe->box));
+  else
+    buf[0] = '\0';
```

Comment:
Is it correct to keep going if either `box` or `sbe` are not set?

How about just setting `buf[0] = '\0'` at the beginning of the function no matter what, then keeping the old check and return if either `buf`, `box`, or `sbe` are NULL?

Reply:
What about padding?

Reply:
What about it? I don’t think you can do much if any of those params are null.

How about just making sure buf is null-terminated and return as early as possible?

Reply:
It turns out that the checks are overkill; none of them can be `NULL`.
There's only one caller and:
- `buf` is `char str[256];`
- `box` is `struct Buffer result = mutt_buffer_make(256);`
- `sbe` is dereferenced before the call: `if (entry->is_hidden)`